packages/block-editor/src/store/selectors.js

```diff
@@ -949,6 +949,14 @@ export function __unstableIsSelectionCollapsed( state ) {
 	);
 }
 
+export function __unstableSelectionHasUnmergeableBlock( state ) {
+	return getSelectedBlockClientIds( state ).some( ( clientId ) => {
+		const blockName = getBlockName( state, clientId );
+		const blockType = getBlockType( blockName );
+		return ! blockType.merge;
```

Comment:
May need to fine tune this by also checking if `switchToBlockType` returns a result.